dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -146,8 +150,12 @@ boolean isSystemClass(String className) {
   SymbolReferenceMaps findSymbolReferences() throws IOException {
     SymbolReferenceMaps.Builder builder = new SymbolReferenceMaps.Builder();
 
+    int count = 0;
     for (ClassPathEntry jar : inputClassPath) {
-      for (JavaClass javaClass : listClasses(jar)) {
+      count++;
+      ImmutableSet<JavaClass> javaClasses = listClasses(jar);
+      System.out.println(count + " : " + jar + ", with " + javaClasses.size() + " classes.");
```

Comment:
For the OutOfMemoryError while reading 570 JAR files, the last output was presto's hadoop.

```
1 : org.apache.beam:beam-sdks-java-io-hcatalog:2.20.0, with 26 classes.
2 : org.apache.beam:beam-vendor-guava-26_0-jre:0.1, with 1951 classes.
3 : org.apache.beam:beam-sdks-java-core:2.20.0, with 3480 classes.
4 : org.apache.beam:beam-sdks-java-io-hadoop-common:2.20.0, with 7 classes.
5 : org.slf4j:slf4j-api:1.7.25, with 34 classes.
6 : com.fasterxml.jackson.core:jackson-annotations:2.10.2, with 69 classes.
7 : com.fasterxml.jackson.core:jackson-core:2.10.2, with 118 classes.
8 : com.fasterxml.jackson.core:jackson-databind:2.10.2, with 656 classes.
9 : org.apache.hadoop:hadoop-common:2.8.5, with 2263 classes.
10 : org.apache.hive:hive-exec:2.1.0, with 17491 classes.
11 : org.apache.hive.hcatalog:hive-hcatalog-core:2.1.0, with 115 classes.
12 : org.apache.beam:beam-model-pipeline:2.20.0, with 607 classes.
13 : org.apache.beam:beam-model-job-management:2.20.0, with 220 classes.
14 : org.apache.beam:beam-vendor-bytebuddy-1_10_8:0.1, with 2359 classes.
15 : com.google.code.findbugs:jsr305:3.0.2, with 35 classes.
16 : org.apache.avro:avro:1.8.2, with 1172 classes.
17 : org.xerial.snappy:snappy-java:1.1.4, with 28 classes.
18 : joda-time:joda-time:2.10.5, with 247 classes.
19 : org.tukaani:xz:1.8, with 116 classes.
20 : junit:junit:4.13-beta-3, with 347 classes.
21 : org.hamcrest:hamcrest-core:2.1, with 1 classes.
22 : org.hamcrest:hamcrest-library:2.1, with 1 classes.
23 : io.airlift:aircompressor:0.16, with 94 classes.
24 : com.facebook.presto.hadoop:hadoop-apache2:3.2.0-1, with 17869 classes.

java.lang.OutOfMemoryError: GC overhead limit exceeded

```

Reply:
The reason why hamcrest-core only has 1 class:
```
suztomo-macbookpro44% jar tf /Users/suztomo/.m2/repository/org/hamcrest/hamcrest-core/2.1/hamcrest-core-2.1.jar 
META-INF/
META-INF/MANIFEST.MF
org/
org/hamcrest/
org/hamcrest/HamcrestCoreIsDeprecated.class
README.txt
```